src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -65,7 +65,13 @@ def iter_path_entries():
                     if entry == "<PATH>":
                         path = self._options_env.get("PATH")
                         if path:
-                            yield from path.split(os.pathsep)
+                            for prefix in path.split(os.pathsep):
+                                if prefix.startswith("$") or prefix.startswith("~"):
```

Comment:
Why do we only care about paths that _start_ with `$` and `~`? I imagine that's the common case, but those characters could easily be embedded within a path too?

Reply:
Trying to hedge here between catching what is almost certainly errors, but *could* be "I know what I'm doing".  At $DAYJOB, like a quarter of the engineers have had a PATH messed up in some way with one of the above prefixes, but if someone really has something like `/usr/$bling/bling$/bin`  I don't want to stop their good time.